subsys/mpsl/cx/bluetooth/mpsl_cx_1w_bluetooth.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file
+ *   This file implements a generic bluetooth external radio 1w coexistence
+ *   interface.
+ */
+#include <stddef.h>
+#include <stdint.h>
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <devicetree.h>
+#include <nrfx.h>
+#include <nrfx_ppi.h>
+#include <nrfx_gpiote.h>
+#include <mpsl_coex.h>
+
+#define COEX_TIMER NRF_TIMER1
+
+#if DT_HAS_COMPAT_STATUS_OKAY(sdc_radio_coex_one_wire)
+#define COEX_NODE DT_COMPAT_GET_ANY_STATUS_OKAY(sdc_radio_coex_one_wire)
+#else
+#define COEX_NODE DT_INVALID_NODE
+#error No "sdc-radio-coex-one-wire"-compatible (and enabled) nodes found
+#endif
+
+#if !IS_ENABLED(CONFIG_SOC_SERIES_NRF52X)
+#error Bluetooth coex is only supported on the nRF52 series
+#endif
+
+#define MPSL_COEX_BT_GPIO_POLARITY_GET(dt_property)                                                \
+	((GPIO_ACTIVE_LOW & DT_GPIO_FLAGS(COEX_NODE, dt_property)) ? false : true)
+
+static volatile bool coex_enabled;
+
+static void coex_enable_callback(void)
+{
+	coex_enabled = true;
+}
+
+int mpsl_cx_bt_interface_1wire_config_set(void)
+{
+	nrfx_err_t err = NRFX_SUCCESS;
+	mpsl_coex_gpiote_cfg_t *gpiote_cfg;
+	nrf_ppi_channel_t ppi_channel;
+	mpsl_coex_if_t coex_if_bt;
+	mpsl_coex_1wire_gpiote_if_t *coex_if = &coex_if_bt.interfaces.coex_1wire_gpiote;
+
+	/* Allocate GPIOTE and PPI channels for the GRANT line */
+	gpiote_cfg = &coex_if->grant_cfg;
+	if (nrfx_gpiote_channel_alloc(&gpiote_cfg->gpiote_ch_id) != NRFX_SUCCESS) {
+		return -ENOMEM;
+	}
+	if (nrfx_ppi_channel_alloc(&ppi_channel) != NRFX_SUCCESS) {
+		return -ENOMEM;
+	}
+	gpiote_cfg->gpio_pin = DT_GPIO_PIN(COEX_NODE, grant_gpios);
```

Comment:
Martí made me aware of a pitfall of using this, it won't work for pins on the Port 1.
Use this instead
```c
NRF_DT_GPIOS_TO_PSEL(COEX_NODE, grant_gpios);
```